cmd/create/cluster/cmd.go

```diff
@@ -722,6 +723,58 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
+	// combine role arns to list
+	roleARNs := []string{
+		roleARN,
+		supportRoleARN,
+		masterRoleARN,
+		workerRoleARN,
+	}
+
+	// parse openshift version
+	openshiftVersion, err := semver.NewVersion(getVersionMinor(version))
+	if err != nil {
+		reporter.Errorf("Could not parse openshift version '%s' : %v", version, err)
+		os.Exit(1)
+	}
+
+	// iterate and validate role arns against openshift version
+	for _, ARN := range roleARNs {
+		// get role from arn
+		role, err := awsClient.GetRoleByARN(ARN)
+		if err != nil {
+			reporter.Errorf("Could not get Role '%s' : %v", ARN, err)
+			os.Exit(1)
+		}
+
+		// get policy openshift version from role tags
+		foundVersion := false
+		for _, tag := range role.Tags {
+			tagValue := awssdk.StringValue(tag.Value)
+			if awssdk.StringValue(tag.Key) == tags.OpenShiftVersion {
+				foundVersion = true
+				policyVersion, err := semver.NewVersion(tagValue)
+				if err != nil {
+					reporter.Errorf("Could not parse openshift policy version from arn '%s' : '%v'", ARN, err)
+					os.Exit(1)
+				}
+
+				// validate policy version is greater than or equal to openshift version
+				if policyVersion.LessThan(openshiftVersion) {
+					reporter.Errorf("Can not proceed with cluster install, policy '%s' version '%s' is"+
+						" less than openshift version '%s'",
+						ARN, policyVersion.String(), openshiftVersion.String())
+					os.Exit(1)
+				}
+			}
+		}
+		// if no version tag is found, error out as we can not guarantee policy's are compatible with openshift version
+		if !foundVersion {
+			reporter.Errorf("Could not validate openshift policy version from arn '%s'", ARN)
+			os.Exit(1)
+		}
```

Comment:
I think we have a function `isRoleCompatible` that does this already. Can you reuse that function (and modify it if necessary), I'd rather have all this code inside `pkg/aws/policies`